Server/ReasnAPI/ReasnAPI/Services/ImageService.cs

```diff
@@ -1,16 +1,145 @@
+            {
+                if (objectType.Name == "User" && newImages.Count > 1)
+                {
+                    context.Images.Add(newImages.First());
+                }
+                else if (objectType.Name == "Event")
+                {
+                    context.Images.AddRange(newImages);
+                }
+            }
+
+            context.SaveChanges();
+        }
+
+        return imageDtos;
+    }
+
+    public ImageDto UpdateImage(int imageId, ImageDto imageDto)
+    {
+        var image = context.Images.FirstOrDefault(r => r.Id == imageId);
+        if (image == null)
+        {
+            return null;
+        }
+
+        image.ObjectId = imageDto.ObjectId;
+        image.ImageData = imageDto.ImageData;
+        image.ObjectTypeId = imageDto.ObjectTypeId;
+
+        context.Images.Update(image);
+        context.SaveChanges();
+        return imageDto;
+    }
+
+    public bool DeleteImage(int id)
+    {
+        var image = context.Images.FirstOrDefault(r => r.Id == id);
+        if (image == null)
+        {
+            return false;
+        }
+        context.Images.Remove(image);
+        context.SaveChanges();
+
+        return true;
+    }
 
-namespace ReasnAPI.Services {
-    public class ImageService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
+    public ImageDto GetImageById(int id)
+    {
+        var image = context.Images.FirstOrDefault(r => r.Id == id);
+        if (image == null)
+        {
+            return null;
+        }
+
+        var imageDto = new ImageDto
+        {
+            ImageData = image.ImageData,
+            ObjectId = image.ObjectId,
+            ObjectTypeId = image.ObjectTypeId
+        };
+
+        return imageDto;
+    }
 
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
+    public IEnumerable<ImageDto> GetAllImages()
+    {
+        var images = context.Images.ToList();
+
+        var imageDtos = images.Select(image => new ImageDto
+        {
+            ImageData = image.ImageData,
+            ObjectId = image.ObjectId,
+            ObjectTypeId = image.ObjectTypeId
+        }).ToList();
+
+        return imageDtos;
     }
-}
+
+    public IEnumerable<ImageDto> GetImagesByFilter(Expression<Func<Image, bool>> filter)
+    {
+        var images = context.Images.Where(filter).ToList();
+
+        var imageDtos = images.Select(image => new ImageDto
+        {
+            ImageData = image.ImageData,
+            ObjectId = image.ObjectId,
+            ObjectTypeId = image.ObjectTypeId
+        }).ToList();
+
+        return imageDtos;
+    }
+
+    public IEnumerable<ImageDto> GetImagesByEventIdAndType(int eventId)
+    {
+        var images = context.Images
+            .Where(image => image.ObjectType.Name == "event" && image.ObjectId == eventId)

```

Comment:
So, it's "Event" or "event"? That's why I said to extract it to const, this way if it's case sensitive - it won't work for one case :)